### PR TITLE
Adventure Chat Integration [EXPERIMENT]

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR is for tracking my experiment. A **lot** of work still needs done. Comments/tests welcome though!

Currently has some breaking changes (in that some color tags parse differently).

This PR depends on the Module API. If we don't end up doing that, I can make changes accordingly.

---
**Target Minecraft Versions:** Any
**Requirements:** N/A
**Related Issues:**
- https://github.com/SkriptLang/Skript/issues/4853
- https://github.com/SkriptLang/Skript/issues/3112
- https://github.com/SkriptLang/Skript/issues/4919 (may not actually be fixed yet)
- https://github.com/SkriptLang/Skript/issues/4951
- https://github.com/SkriptLang/Skript/issues/4952 (double check to make sure it's fixed)

### CODE REVIEW

File: src/main/resources/lang/default.lang [hunk 1]
```diff
@@ -1946,3 +1946,7 @@ types:
 	# Hooks
 	money: money
 	region: region¦s
+
+	# Other API
+
+	component: component¦s @a
```
Comment (TheLimeGlass) [inline src/main/resources/lang/default.lang:1952]: I think we should add chat component as another aliases

File: src/main/java/org/skriptlang/skript/bukkit/chat/util/CodeConverter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		CODE_MAP.put('e', "yellow");
+		CODE_MAP.put('f', "white");
+
+		CODE_MAP.put('o', "italic");
+		CODE_MAP.put('l', "bold");
+		CODE_MAP.put('m', "strikethrough");
+		CODE_MAP.put('n', "underlined");
+		CODE_MAP.put('k', "obfuscated");
+
+		CODE_MAP.put('r', "reset");
```
Comment (kiip1) [inline src/main/java/org/skriptlang/skript/bukkit/chat/util/CodeConverter.java:54]: Pull from `org.bukkit.ChatColor` instead, except for underlined and obfuscated. (For those the Enum name doesn't match this string one).

File: src/main/java/org/skriptlang/skript/bukkit/chat/util/ComponentHandler.java [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+	 * Constructs an audience from command senders.
+	 * @param senders The members of this audience.
+	 * @return An audience consisting of the provided command senders.
+	 */
+	public static Audience audienceFrom(CommandSender... senders) {
+		List<Audience> bukkitAudiences = new ArrayList<>();
+		for (CommandSender sender : senders)
+			bukkitAudiences.add(getAdventure().sender(sender));
+		return Audience.audience(bukkitAudiences);
+	}
```
Comment (kiip1) [inline src/main/java/org/skriptlang/skript/bukkit/chat/util/ComponentHandler.java:347]:
```suggestion
	public static Audience audienceFrom(CommandSender... senders) {
		return audienceFrom(ImmutableList.copyOf(senders));
	}
```

File: src/main/java/org/skriptlang/skript/bukkit/chat/util/ComponentHandler.java [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+	/**
+	 * Constructs an audience from command senders.
+	 * @param senders The members of this audience.
+	 * @return An audience consisting of the provided command senders.
+	 */
+	public static Audience audienceFrom(Collection<CommandSender> senders) {
+		List<Audience> bukkitAudiences = new ArrayList<>();
+		for (CommandSender sender : senders)
+			bukkitAudiences.add(getAdventure().sender(sender));
+		return Audience.audience(bukkitAudiences);
```
Comment (kiip1) [inline src/main/java/org/skriptlang/skript/bukkit/chat/util/ComponentHandler.java:334]: Could use Stream api but this is fine.

File: src/main/java/org/skriptlang/skript/bukkit/chat/util/ComponentHandler.java [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ComponentHandler {
```
Comment (kiip1) [inline src/main/java/org/skriptlang/skript/bukkit/chat/util/ComponentHandler.java:48]:
```suggestion
@ApiStatus.Internal
public final class ComponentHandler {
	
	private ComponentHandler() {}
```

**Comment by APickledWalrus** [comment]
Closing as I do not feel these changes are well integrated. I'll re-examine how we can better integrate these libraries with a fresh start.

**Pull request closed without merging**